### PR TITLE
Force HTTPS if the original port is 443 behind a reverse proxy

### LINKED ISSUE #1022: Notes links use 443 port
Hello,
I have a trouble in my Shaarli as you can see it here : https://husimo.fr/shaarli/?ireqPQ
The note link is http and then you have the 443 port displayed in the URL. But my browser doesn't like it so these links are not working.
I think it may be related to the fact I'm using OVH provided for hosting (free offer with domain name with 10MB space, just what I need for Shaarli !), with SSL Let's encrypt certificate.
I have the last version of Shaarli.
Thank you

### PR DESCRIPTION
**Pull request by ArthurHoaro**
Fixes #1022

### CODE REVIEW

File: application/HttpUtils.php [hunk 1]
```diff
@@ -302,6 +302,10 @@ function server_url($server)
                 $port = $server['HTTP_X_FORWARDED_PORT'];
             }
 
+            if ($port == '443') {
```
Comment (virtualtam) [inline application/HttpUtils.php]: I'm not sure we should provide support for messed up HTTP server configs, especially given how easy to get things wrongs when chaining proxies...

maybe an error message to tell the user there's something wrong with the HTTP server configuration would be better / more relevant?

Comment (nodiscc) [reply application/HttpUtils.php]: I tend to agree with you , on the other hand in cases like https://github.com/shaarli/Shaarli/issues/1022 the user simply can't fix the configuration by themselves (shared hosting...). This is a simple fix (with a test!) so I'd be ok with merging it, however "we don't provide support for messed up HTTP server configs" sounds like a good guideline in general.

Comment (ArthurHoaro) [reply application/HttpUtils.php]: Well, @nodiscc said what I think.

Also, we have this behaviour for non proxy configuration. And I don't think that any browser would let you connect to the port `443` over HTTP.

Comment (virtualtam) [reply application/HttpUtils.php]: Works for me, would you mind adding a comment to highlight this as a server setup workaround?

Comment (ArthurHoaro) [reply application/HttpUtils.php]: Sure.

**Pull request merged**